### DISCUSSION TITLE
Running ETS2 Linux native with steamcmd

### DISCUSSION THREAD

**Discussion by kallestrop** [truckersmp-cli/truckersmp-cli#339] [Q&A]
As your truckersmp-cli script is using steamcmd, maybe someone can give me a hint with this issue.

I have installed Linux native version of ETS2 in my Steam client. To avoid a full download every time i switch between beta and stable branch, i made a second install of the beta with steamcmd. 

The installation itself seems to be fine, but i always get an error message when trying to run the game with `app_run [Steamid]`

Steps i made:

```
Steam>force_install_dir /media/ssd_4tb/user/ets2_beta/ets2beta

Steam>login steamuser

Logging in user 'steamuser' to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK

Steam>app_update 227300 -beta public_beta validate
 Update state (0x11) preallocating, progress: 0.00 (0 / 23080941309)
 Update state (0x61) downloading, progress: 0.05 (12413386 / 23080941309)
.
.
 Update state (0x81) verifying update, progress: 91.94 (21219897892 / 23080941309)
 Update state (0x81) verifying update, progress: 98.23 (22671554033 / 23080941309)
Success! App '227300' fully installed.

Steam>app_run 227300
ERROR! AppId 227300 launch failed (Missing executable)
```

**Comment by Lucki** [comment]
AFAIK steamcmd can't run games, at least those that have DRM.

Some options I can think of:
* Try to fool the steam client by shuffling the data around with symlinks. And maybe editing the corresponding `.acf` file so the files won't update to the wrong branch.
* Use an alternative client which might have such functionality. It needs to implement the `steamclient.so` for the DRM to work. E.g. `steam-tui` does not but `opensteamclient` does. But I haven't looked into these in depth.

**Comment by kallestrop** [comment]
Thanks for the reply. I will investigate this topic further when the next beta arrives. Seems that we get 1.48 stable in a few days. 

But beside that, i wonder why i can run the game with steamcmd in Windows.

 I just installed the beta the same way in my Windows 10 install, and it's running correctly:
 
 ```
Steam>app_run 227300
src\public\tier1\interface.h (271) : m_pModule
src\public\tier1\interface.h (271) : m_pModule
Failed to launch 64-bit injection helper
Game process added : AppID 227300 "c:\ets2_148beta\bin\win_x64\eurotrucks2.exe", ProcID 11152, IP 0.0.0.0:0
Game process removed: AppID 227300 "c:\ets2_148beta\bin\win_x64\eurotrucks2.exe", ProcID 11152
```
